src/utilities.lisp

```diff
@@ -0,0 +1,24 @@
+;;;; utilities.lisp
+;;;;
+;;;; Author: Eric Peterson
+
+(in-package #:rpcq)
+
+
+(defun sanitize-name (symb)
+  "Convert a stringifiable SYMB into a snake-case string."
+  (substitute #\_ #\- (string symb)))
+
+(defun str->lisp-keyword (str)
+  "Convert a snake-case string into a hyphenated Lisp keyword."
+  (make-keyword (substitute #\- #\_ (string-upcase str))))
+
+(defun format-log (s fmt-string &rest args)
+  "Writes a format string to the stream S in debug output format."
+  (format s "[~A | ~A] ~?" (local-time:now) (bt:thread-name (bt:current-thread)) fmt-string args)
```

Comment:
```suggestion
  (format s "[~A | ~A] ~?~%" (local-time:now) (bt:thread-name (bt:current-thread)) fmt-string args))
```

Reply:
cool: i wasn't sure how intelligent the arg parsing for `~?` was, and i wasn't brave enough to just try it.